crates/e2e-test-utils/src/testsuite/actions.rs

```diff
@@ -0,0 +1,383 @@
+//! Actions that can be performed in tests.
+
+use crate::testsuite::{BoxFuture, Environment};
+use alloy_primitives::{BlockNumber, B256, U256};
+use alloy_rpc_types_engine::{ExecutionPayload, PayloadAttributes};
+use eyre::Result;
+use std::sync::Arc;
+
+/// An action that can be performed on an instance.
+///
+/// Actions can produce results that are stored for later assertions.
+pub trait Action<I>: Send + 'static {
+    /// Executes the action
+    fn execute<'a>(&'a mut self, env: &'a Environment<I>) -> BoxFuture<'a, Result<ActionResult>>;
```

Comment:
should this consume self?

or is there are reason we want to persist actions?

Reply:
yep, i had issues with trait objects, I've pushed a workaround to actually consume self, ptal